### PR TITLE
fix: Show "interesting" doc relations on HTMLized page

### LINKED ISSUE #5344: /doc/html replaced drafts
### Description

A replaced draft on /doc shows what it was replaced with.

A draft that replaces another shows what it replaced on /doc/html.  That is fine, though it only does so via the "Compare versions" feature.

A draft that has been replaced does not show what replaced it anywhere on /doc/html.

It seems like this might be a useful feature when looking at a draft, especially for drafts that have been adopted, where you might want to see the latest.

Story: I was trying to find my way from https://datatracker.ietf.org/doc/html/draft-xu-risav to the latest version.  I ended up going to GitHub (which the draft conveniently links) to find what replaced the draft (though the /doc link might have been faster).

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by larseggert**
Fixes #5344

### CODE REVIEW

File: ietf/doc/views_doc.py [hunk 1]
```diff
@@ -139,20 +139,17 @@ def render_document_top(request, doc, tab, name):
 def interesting_doc_relations(doc):
 
     if isinstance(doc, Document):
-        cls = RelatedDocument
```
Comment (rjsparks) [inline ietf/doc/views_doc.py:142]: This is here for a reason?
You will report the wrong relations if you are looking at an older version of a document if you take this out.

**Comment by rjsparks** [comment]
Holding this for discussion at codesprint

**Comment by rjsparks** [comment]
Bah - we missed talking about this Saturday - help watch for a chance to talk about it while here this week

**Comment by rjsparks** [comment]
We should talk about this during 117.

**Review by Limpisey77922321 — COMMENTED** [review]
[Build and Release](https://github.com/ietf-tools/datatracker/actions/workflows/build.yml)

**Comment by larseggert** [comment]
Converting to draft until we have a chance to talk.

**Comment by rjsparks** [comment]
closing this as asleep - it can be pulled back out later if needed.

**Pull request closed without merging**